### PR TITLE
Access Control on Publishing detectors

### PR DESCRIPTION
**Pull request by ShekharGupta1988**
If users doesnt have access to publishing detectors(imposed by owning teams), they will see the following notification:

image

### CODE REVIEW

File: ApplensBackend/Controllers/DiagnosticController.cs [hunk 1]
```diff
@@ -189,6 +207,42 @@ public async Task<IActionResult> Invoke([FromBody]JToken body)
...
+            {
+                groupNames = resourceConfig.AllowedGroupsToPublish.Select(p => p.Name).ToList();
+            }
+
+            return Ok(new
+            {
+                HasAccess = hasAccess,
+                ServiceName = resourceConfig?.Name,
+                ResourceType = resourceConfig?.ResourceType,
+                ResourceOwners = resourceConfig?.ResourceOwners,
```
Comment (khaled-zayed) [inline ApplensBackend/Controllers/DiagnosticController.cs:240]: Does that mean that teams can add a new property in the enabledResourceTypes json file called ResourceOwners and the value is the security group alias?

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html [hunk 2]
```diff
@@ -94,6 +94,41 @@ <h4 style="font-weight: bold; color: #373277">Local Development Instructions</h4
...
+      <div class="modal-title-container">
+        <h4 class="modal-title">Access Denied</h4>
+        <span class="modal-sub-title">Unable to Publish Public Detectors</span>
+      </div>
+    </div>
+    <div class="modal-content">
+     
+      <div class="content-row">
+        <span>
+          The following team have restricted access to publish public-facing detectors.
```
Comment (khaled-zayed) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html:109]: tiny correct have => has. Everything else looks good, we're going to have some happy partners 😊

File: ApplensBackend/Controllers/DiagnosticController.cs (hunk 1 above)
Comment (ShekharGupta1988) [reply ApplensBackend/Controllers/DiagnosticController.cs:240]: Nope. These settings cannot be in enabledResourceTypes.json due to compliance and security reasons.
So these settings are saved in our cosmos DB server. While working on detector graduation proposal, lets discuss more to streamline multiple partner configs at one place which is secure and compliant

The idea is to have :
- Owners for every service (for general communication/admin operations in future)
- For  publishing Access Control:
      - List of users who can publish and/or
      - List of Groups who can publish

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html (hunk 2 above)
Comment (ShekharGupta1988) [reply AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html]: fixed

File: ApplensBackend/Helpers/Utilities.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+        /// <summary>
+        /// Validates if a user belongs to a security or distribution group.
+        /// </summary>
+        /// <param name="userAlias">user Alias</param>
+        /// <param name="groupObjectId">group object id.</param>
+        /// <returns>True, if user is part of the group.</returns>
+        public static async Task<bool> CheckUserGroupMembership(string userAlias, string groupObjectId)
+        {
+            string graphUrl = "https://graph.microsoft.com/v1.0/users/{0}/checkMemberGroups";
```
Comment (cindylovescoding) [inline ApplensBackend/Helpers/Utilities.cs:44]: Nitpick: we can use the graph endpoint constants from Constants.cs

Comment (ShekharGupta1988) [reply ApplensBackend/Helpers/Utilities.cs]: fixed

File: ApplensBackend/Services/CosmosDBHandler/CosmosDBHandler.cs [hunk 4]
```diff
@@ -45,9 +54,9 @@ public async Task<T> GetItemAsync(string id)
                 {
                     if (e.Message.Contains("Resource Not Found"))
                     {
-                        CreateDatabaseIfNotExistsAsync().Wait();
-                        CreateCollectionIfNotExistsAsync().Wait();
-                        return await GetItemAsync(id);
+                        await CreateDatabaseIfNotExistsAsync();
```
Comment (cindylovescoding) [inline ApplensBackend/Services/CosmosDBHandler/CosmosDBHandler.cs:57]: Just curious, is there any dependency between CreateDatabaseIfNotExistsAsync() and CreateCollectionIfNotExistsAsync()?

Comment (ShekharGupta1988) [reply ApplensBackend/Services/CosmosDBHandler/CosmosDBHandler.cs:57]: Yup. The collection is inside the cosmos database. So need to create that first followed by collection

**Review by cindylovescoding — APPROVED** [review]

**Review by SharmaAjay19 — APPROVED** [review]
Perfect! Love the generalization and refactor of old code.

**Review by XiaoxuHU — APPROVED** [review]

**Pull request merged**